dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -524,29 +535,30 @@ private boolean createWorkflowsAndVersionsFromDockstoreYml(List<? extends Workfl
                         addValidationsToMessage(workflow, version, messageWriter);
                         publishWorkflowAndLog(workflow, publish, user, repository, gitReference);
                     });
+                    successfulWorkflowNames.add(computeWorkflowName(wf));
+                } catch (RuntimeException | DockstoreYamlHelper.DockstoreYamlException ex) {
+                    // If there was a problem updating the workflow (an exception was thrown), either:
+                    // a) rethrow certain exceptions to abort .dockstore.yml parsing, or
+                    // b) log something helpful and move on to the next workflow.
+                    isSuccessful = false;
+                    if (ex instanceof RuntimeException) {
+                        rethrowIfFatal((RuntimeException)ex, transactionHelper);
+                    }
+                    final String message = String.format("Failed to create %s '%s':\n- %s\n",
+                        computeTermFromClass(workflowType), computeWorkflowName(wf), generateMessageFromException(ex));
+                    LOG.error(message, ex);
+                    messageWriter.println(message);
+                    failedWorkflowNames.add(computeWorkflowName(wf));
                 }
-            } catch (RuntimeException | DockstoreYamlHelper.DockstoreYamlException ex) {
-                // If there was a problem updating the workflow (an exception was thrown), either:
-                // a) rethrow certain exceptions to abort .dockstore.yml parsing, or
-                // b) log something helpful and move on to the next workflow.
-                isSuccessful = false;
-                if (ex instanceof RuntimeException) {
-                    rethrowIfFatal((RuntimeException)ex, transactionHelper);
-                }
-                final String message = String.format("Error processing %s %s:%n%s",
-                    computeTermFromClass(workflowType), computeFullWorkflowName(wf.getName(), repository), generateMessageFromException(ex));
-                LOG.error(message, ex);
-                messageWriter.println(message);
-                messageWriter.println("Entry skipped.");
             }
         }
 
-        return isSuccessful;
+        return createBasicEvent(repository, gitReference, username, isSuccessful ? successfulWorkflowNames : failedWorkflowNames, LambdaEvent.LambdaEventType.PUSH, isSuccessful);
```

Comment:
Hmmmm, don't care about the extra events (I think postgres can easily handle it). 
If it makes the code simpler (the whole combining thing is why I took a note above and the UI simpler, I'm for it. 
We could event throw in some modest search like the browse pages https://dockstore.org/tools

Reply:
Agreed that it shouldn't be an issue from the DB side. And the API calls for the events all look paginated, so that's good. And it does make logical sense to have a one-to-one correspondence.

The UI will be interesting, at least for outliers. The [viral-pipelines](https://github.com/broadinstitute/viral-pipelines/blob/master/.dockstore.yml) repo has 72 entries, so if you're looking for the next-to-last event, there will be a bit of paging. :)

Reply:
One idea would be to also add something like a "group" id to each Event, where a "group" would be the Events that were created in response to a particular logical [webhook] event.  We might not use it initially, but later, it could be useful to more gracefully format repos with large numbers of entries.